rules/prefer-error-cause.js

```diff
@@ -0,0 +1,297 @@
+'use strict';
+const esquery = require('esquery');
+const {isClosingBraceToken, isCommaToken} = require('eslint-utils');
+const {matches, methodCallSelector} = require('./selectors/index.js');
+const {replaceReferenceIdentifier} = require('./fix/index.js');
+
+const ERROR = 'error';
+const SUGGESTION = 'suggestion';
+
+const messages = {
+	[ERROR]: 'Prefer to add the old error as new error\'s `cause` property.',
+	[SUGGESTION]: 'Add the old error as new error\'s `cause` property.',
+};
+
+const tryCatchSelector = matches([
+	[
+		'CatchClause',
+		':has(ThrowStatement)',
+	].join(''),
+]);
+
+const promiseCatchSelector = matches([
+	[
+		methodCallSelector({method: 'catch', argumentsLength: 1}),
+		':has(ThrowStatement)',
+	].join(''),
+]);
+
+const peek = array => {
+	if (!array || array.length === 0) {
+		return undefined;
+	}
+
+	return array[array.length - 1];
+};
+
+// Same logic with '/rules/fix/append-argument.js'
+// But the function does not support NewExpression.
+const appendArgument = (fixer, node, text, sourceCode) => {
```

Comment:
In your case, you only have `NewExpression` with at least one argument, you can update that check logic in /rules/fix/append-argument.js to only throw when `if (node.type === "NewExpression" && node.arguments.length === 0)`.(I hope you know why that check exists). So we can use that here.